File: Data Structures & Algorithms/valid-palindrome-ii/submission-3.py
class Solution:
    def validPalindrome(self, s: str) -> bool:
        isUseJump = False

        l, r = 0, len(s)-1

        def checkLoop(left, right):
            while left < right:
                if s[left] == s[right]:
                    left += 1
                    right -= 1
                    continue
                return False
            return True

        while l < r:
            if s[l] == s[r]:
                l += 1
                r -= 1
                continue

            return checkLoop(l+1, r) or checkLoop(l, r-1)
        return True

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-3.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    findMin(nums) {
        let left = 0, right = nums.length-1;
        let mid = left + Math.floor((right-left)/2)
        while (left <= right) {
            if (nums[left] > nums[mid] && nums[right] > nums[mid]) {
                right = mid;
            } else if (nums[left] < nums[mid] && nums[right] < nums[mid]) {
                left = mid;
            }
            mid = left + Math.floor((right-left)/2);
            if (nums[left] >= nums[mid] && nums[mid] >= nums[right]) {
                return nums[right]
            } else if (nums[left] <= nums[mid] && nums[mid] <= nums[right]){
                return nums[left]
            }
        }
    }
}
